### PR TITLE
The scalar field value of search result is wrong after deleted

### LINKED ISSUE #4200: The scalar field information of search result is wrong after some entities deleted
**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
I'm trying to use milvus_cpu_0.11.0 as my embedding search engine. I save each embedding with its class_id.
I first add 100 random vectors with its id range [0 to 99].
Then I search for top-3 of first vector, it works fine. First result is the embedding with ID "0" and class_id "0". All others id align with class_id.
```
Results: 
- id: 0
- distance: 0.0
- class_id: 0
- id: 85
- distance: 72.66512298583984
- class_id: 85
- id: 74
- distance: 75.49613952636719
- class_id: 74
```

However, when I delete first vector and search again, Milvus returns nonsense result:
The first result: vector id "0" with class_id "85" and the distance is 0.0 ??!? Does it shuffle the embedding ID and not delete it, the class_id is also shuffled.
The second result: As you can see there are class_id 1241513984 which is out of class_id range of [0 to 99] ??!?
The third result: the first vector is deleted, but it is still in the search result but with a distance of 75.49613952636719 ??!?
```
Results after delete first vector: 
- id: 0
- distance: 0.0
- class_id: 85
- id: 85
- distance: 72.66512298583984
- class_id: 1241513984
- id: 74
- distance: 75.49613952636719
- class_id: 0
```

The code to reproduce:
```
import random
import numpy as np
from pprint import pprint
from milvus import Milvus, DataType
​
random.seed(0)
np.random.seed(0)
# ------
# Setup:
#    First of all, you need a running Milvus(0.11.x). By default, Milvus runs on localhost in port 19530.
#    Then, you can use pymilvus(0.3.x) to connect to the server, You can change the _HOST and _PORT accordingly.
# ------
_HOST = '127.0.0.1'
_PORT = '19530'
client = Milvus(_HOST, _PORT)
​
collection_name = "face_emb"
partition_tag = 'test'
​
# Returns the version of the client.
client_version= client.client_version()
print(client_version)
​
# Returns the version of the Milvus server.
server_version = client.server_version(timeout=10)
print(server_version)
​
# Create collection
if collection_name in client.list_collections():
    client.drop_collection(collection_name)
collection_param = {
    "fields": [
        #  Milvus doesn't support string type now, but we are considering supporting it soon.
        #  {"name": "title", "type": DataType.STRING},
        {"name": "class_id", "type": DataType.INT64},
        {"name": "embedding", "type": DataType.FLOAT_VECTOR, "params": {"dim": 512}},
    ],
    "segment_row_limit": 4096,
    "auto_id": False
}
​
client.create_collection(collection_name, collection_param)
client.create_partition(collection_name, partition_tag)
collection = client.get_collection_info(collection_name)
pprint(collection)
​
# Create sample data
ids =  list(range(100))
embeddings = [np.random.random((512)) for _ in range(100)]
​
# Insert vectors
hybrid_entities = [
    {"name": "class_id", "values": ids, "type": DataType.INT64},
    {"name": "embedding", "values": embeddings, "type": DataType.FLOAT_VECTOR},
]
ids = client.insert(collection_name, hybrid_entities, ids, partition_tag=partition_tag)
before_flush_counts = client.count_entities(collection_name)
client.flush([collection_name])
after_flush_counts = client.count_entities(collection_name)
print("\n----------flush----------")
print("There are {} faces in collection `{}` before flush".format(before_flush_counts, collection_name))
print("There are {} faces in collection `{}` after flush".format(after_flush_counts, collection_name))
​
query_hybrid = {"bool": {"must": [
    {"vector": {"embedding": {"topk": 3, "query": embeddings[:1], "metric_type": "L2"}}}
]}}
​
# Search
results = client.search(collection_name, query_hybrid, fields=["class_id"])
for entities in results:
    print("\nResults: ")
    for topk_face in entities:
        print("- id: {}".format(topk_face.id))
        print("- distance: {}".format(topk_face.distance))
        print("- class_id: {}".format(topk_face.entity.class_id))
​
# Delete first vector
client.delete_entity_by_id(collection_name, ids=[0])
client.flush([collection_name])
print("\nDetele first vector\n")
​
# Query again to check the vector is deleted
query_hybrid = {"bool": {"must": [
    {"vector": {"embedding": {"topk": 3, "query": embeddings[:1], "metric_type": "L2"}}}
]}}
​
results = client.search(collection_name, query_hybrid, fields=["class_id"])
for entities in results:
    print("Results after delete first vector: ")
    for topk_face in entities:
        print("- id: {}".format(topk_face.id))
        print("- distance: {}".format(topk_face.distance))
        print("- class_id: {}".format(topk_face.entity.class_id))
```

**Steps/Code to reproduce behavior**
Follow this [guide](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

**Expected behavior**

**Method of installation**
- [x] Docker/cpu
- [ ] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Softwars conditions (OS, CPU, GPU, Memory)

- Milvus version: 0.11.0

- Milvus SDK version: python sdk -- pymilvus-0.3.0

**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by BossZou**
Signed-off-by: Yhz <yinghao.zou@zilliz.com>

**What type of PR is this?**

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

Fixes #4200, #4256

**What this PR does / why we need it:**

Fix bug.

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: core/src/server/delivery/request/SearchReq.cpp [hunk 1]
```diff
@@ -99,7 +99,7 @@ SearchReq::OnExecute() {
                 }
 
                 // validate search metric type and DataType match
-                bool is_binary = (field->GetFtype() == engine::DataType::VECTOR_FLOAT) ? false : true;
+                bool is_binary = !(field->GetFtype() == engine::DataType::VECTOR_FLOAT);
```
Comment (scsven) [inline core/src/server/delivery/request/SearchReq.cpp:102]: Simplified to `!=`

**Review by wxyucs — CHANGES_REQUESTED** [review]

File: core/src/server/delivery/request/SearchReq.cpp (hunk 1 above)
Comment (BossZou) [reply core/src/server/delivery/request/SearchReq.cpp]: Fixed.

**Review by wxyucs — DISMISSED** [review]

File: core/src/server/delivery/request/SearchReq.cpp [hunk 2]
```diff
@@ -99,7 +99,7 @@ SearchReq::OnExecute() {
                 }
 
                 // validate search metric type and DataType match
-                bool is_binary = (field->GetFtype() == engine::DataType::VECTOR_FLOAT) ? false : true;
+                bool is_binary = field->GetFtype() != engine::DataType::VECTOR_FLOAT;
```
Comment (yhmo) [inline core/src/server/delivery/request/SearchReq.cpp]: bool is_binary = (field->GetFtype() != engine::DataType::VECTOR_FLOAT);

**Review by yhmo — APPROVED** [review]

File: core/src/server/delivery/request/SearchReq.cpp (hunk 2 above)
Comment (BossZou) [reply core/src/server/delivery/request/SearchReq.cpp]: Changed.

**Review by cydrain — APPROVED** [review]

**Review by wxyucs — APPROVED** [review]

**Pull request merged**